src/api/app/models/project.rb

```diff
@@ -968,6 +976,44 @@ def branch_to_repositories_from(project, pkg_to_enable, opts = {})
     end
   end
 
+  def branch_remote_repositories(project)
+    remote_project = RemoteProject.new(name: project)
+    remote_project_meta = Nokogiri::XML(remote_project.meta.to_s)
+    local_project_meta = Nokogiri::XML(to_axml)
+
+    remote_repositories = remote_project.repositories_from_meta
+    remote_repositories -= repositories.where(name: remote_repositories).pluck(:name)
+
+    remote_repositories.each do |repository|
+      repository_node = local_project_meta.create_element("repository")
+      repository_node["name"] = repository
+
+      # if it is kiwi type
+      if repository == "images"
+        path_element = remote_project_meta.xpath("//repository[@name='images']/path")
```

Comment:
Did you try what happens if this is ``nil``?

Actually it is path_elements but otherwise you would also need to rename the other variables ... nitpick...